plugins/inputs/t128_tank/reader.go

```diff
@@ -118,6 +118,13 @@ func NewReader(tankAddress string, tankPort int, topic string, indexPath string,
 		log:            log,
 		acc:            acc,
 	}
+	lastSavedIndex, err := r.getIndex(indexPath, defaultIndex)
```

Comment:
This is not the correct place to identify the last saved index. That should be part of `Run` (and it already is...).